### ISSUE TITLE
Crash on exit?

### ISSUE THREAD

**Issue by alexiskennedy** [weatherfactory/cultistsimulator-visible#223]
[hs7156]

Fun one for you: heavily modded game crashes on exit (may or may not have anything to do with the mods).

[Attachments.zip](https://github.com/weatherfactory/cultistsimulator-visible/files/9900254/Attachments.zip)

**Comment by Chelnoque** [comment]
That's actually a known - famous, even -  problem. Any kind of patching makes the game to crash on exit. No idea why, all error messages are absolute 🥴  
But, well, better on exit than on startup, right? Considering the harmlessness and enigmatic nature, think it's better to leave it be.

**Comment by alexiskennedy** [comment]
'Any kind of patching' - you mean if the game files are changed while the exe is running? or something else?

**Comment by Chelnoque** [comment]
If the game *code* was modified. Well, to be precise "if the game code was modified by [Harmony](https://github.com/pardeike/Harmony/releases), which is used the most for Unity DLL modding, and the only thing I've tested". Not even sure whether there are other convenient means to do that.
Anyways, that means, roughly "DLL patch = the game shows an error on exit".

**Comment by alexiskennedy** [comment]
Well if this is so, and this is the first time someone's reported a bug, it's clearly not a critical problem. It does make me uneasy in my soul though. How aggressively does the error manifest? Is it a n OS dialog box warning? not a cat screen presumably...?

**Comment by Chelnoque** [comment]
No, it's pretty non-intrusive overral - you press exit, game closes, but there's a brief whoops crash Unity window with a progres bar. Doesn't even suggest sending a message like usual Unity engine crashes. But still annoying.

Fun thing that I suddenly can't reproduce it on k.4. Maybe the DLL tangle somehow got cut. Or maybe it's a dice roll when the build is compiled.

**Comment by alexiskennedy** [comment]
There's a really weird crash I used to get years ago with a similar manifestation. It used to happen sometimes if I launched the game with Steam running, exited and then restarted inside about forty seconds. I don't think I ever saw it reported by users, so I don't know if it even happened on machines without the Unity editor installed (or at my latitude, or whatever the hell).

I doubt it's related, but I do wonder whether Steam is implicated somehow?

**Comment by alexiskennedy** [comment]
(anyway happy to leave as pri 8 for now)
